src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -234,29 +236,22 @@ int main(int argc, char* argv[])
         simulatorOutputStream = &simulationOutputFileStream;
     }
 
-    // Redirect the Q# operation output from std::cout if the --operation-output option is present.
-    ostream* operationOutputStream = &cout;
-    ofstream operationOutputFileStream;
-    if (!operationOutputFileOpt->empty())
-    {
-        operationOutputFileStream.open(operationOutputFile);
-        operationOutputStream = &operationOutputFileStream;
-    }
-
     // Run simulation and write the output of the operation to the corresponding stream.
-    int64_t operationOutput = Quantum__StandaloneSupportedInputs__ExerciseInputs__body(
-        intValue, doubleValue, result, qirString);
+    Quantum__StandaloneSupportedInputs__ExerciseInputs(
+        intValue,
+        integerArray.get(),
+        doubleValue,
+        doubleArray.get(),
+        boolAsCharValue,
+        boolArray.get(),
+        pauliAsCharValue,
+        pauliArray.get(),
+        rangeValue.get(),
+        resultAsCharValue,
+        resultArray.get(),
+        stringValue.c_str());
 
     simulatorOutputStream->flush();
-    (*operationOutputStream) << operationOutput << endl;
-    operationOutputStream->flush();
-
-    // Close opened file buffers;
-    if (operationOutputFileStream.is_open())
-    {
-        operationOutputFileStream.close();
-    }
-
     if (simulationOutputFileStream.is_open())
     {
         simulationOutputFileStream.close();
```

Comment:
**Bug?**
Look at the following sequence:

```c++
// (1) The vector is created:
vector<int64_t> integerVector;
// (2) The vector is filled in:
app.add_option(.., integerVector, ..)..
```
(3) Then the call happens:
unique_ptr\<InteropArray> integerArray = CreateInteropArray(**integerVector**);

(4) The function called has the following signature:
unique_ptr<InteropArray> CreateInteropArray(**vector\<T> v**)

The parameter `v` is passed _by value_ (as opposed to "by reference"). The parameter `v` is created as a copy of the argument **integerVector**. The parameter `v` is similar to a local variable of the function `CreateInteropArray()`. It has a pointer to the heap-allocated block. That pointer can be acquired with the call `v.data()`.

Now here is what happens in the function:
```c++
unique_ptr<InteropArray> CreateInteropArray(vector<T> v)
{
    unique_ptr<InteropArray> array(new InteropArray());
    array->Size = v.size();
    array->Data = v.data();   // Fragment of interest.
    return array;
}   // Fragment of interest.
```
The fragment `array->Data = v.data()` memorizes the pointer to the heap block.
At the end of the function `}` the parameter `v` goes out of the scope and is destroyed, its destructor deallocates the heap-allocated block. The `array->Data` becomes a dangling pointer, and the local variable `array` (containing such a dangling pointer) is returned. 

After the call (3) the local variable `integerArray` contains a dangling pointer.

Possible solution: pass _by reference_ rather than _by value_. I.e. change the signature to 
unique_ptr\<InteropArray> CreateInteropArray(vector\<T>**&** v)
(the ampersand `&` is added in front of the parameter name `v`).

Reply:
Great catch! I'll change the signature and make sure that we exercise a read to data to make sure the values are correct.

Reply:
Updated the signature and added additional Q# code that prints the values of array to verify execution correctness.